### PR TITLE
add actions button to the video miniature

### PR DESCRIPTION
**Pull request by rigelk**
This PR adds an action menu for quick access to a few common actions one could wish to do from any video listing.

### CODE REVIEW

File: client/src/app/shared/video/video-miniature.component.html [hunk 1]
```diff
@@ -14,12 +14,81 @@
     </a>
 
     <span i18n class="video-miniature-created-at-views">{{ video.publishedAt | myFromNow }} - {{ video.views | myNumberFormatter }} views</span>
+  
+    <div class="dropdown" ngbDropdown placement="bottom-right" role="group">
+
+      <button class="icon-action" aria-label="Actions on this video" title="Actions on this video" 
+              id="moreOptions" data-flip="true"
```
Comment (Chocobozzz) [inline client/src/app/shared/video/video-miniature.component.html]: What is the purpose of `data-flip`?

and `id="more-options"`

File: client/src/app/shared/video/video-miniature.component.html [hunk 2]
```diff
@@ -14,12 +14,81 @@
...
+              (click)="loadDropdownInformations()"
```
Comment (Chocobozzz) [inline client/src/app/shared/video/video-miniature.component.html]: Maybe `mouseover`?

File: client/src/app/shared/video/video-miniature.component.html [hunk 3]
```diff
@@ -14,12 +14,81 @@
...
+
+      <div class="dropdown-menu" ngbDropdownMenu>
+
+        <!-- User common actions -->
+        <button class="dropdown-item" i18n-title title="Un/subscribe to this channel" (click)="toggleSubscription()">
+          <span *ngIf="subscribed !== true" i18n>Subscribe to {{ video.channel.displayName }}</span>
+          <span *ngIf="subscribed === true" i18n>Unsubscribe from {{ video.channel.displayName }}</span>
+        </button>
+
+        <a *ngIf="isVideoUpdatable()" i18n-title title="Update this video" class="dropdown-item" href="#" [routerLink]="[ '/videos/update', video.uuid ]">
```
Comment (Chocobozzz) [inline client/src/app/shared/video/video-miniature.component.html]: remove `href="#"`

File: client/src/app/shared/video/video-miniature.component.html [hunk 4]
```diff
@@ -14,12 +14,81 @@
...
+          <ng-container i18n>Update</ng-container>
```
Comment (Chocobozzz) [inline client/src/app/shared/video/video-miniature.component.html]: You don't need `ng-container`

File: client/src/app/shared/video/video-miniature.component.html [hunk 5]
```diff
@@ -14,12 +14,81 @@
...
+        </a>
+
+        <button class="dropdown-item" i18n-title title="Delete this video" *ngIf="isVideoRemovable()" (click)="removeVideo($event)">
+          <span i18n>Delete</span>
```
Comment (Chocobozzz) [inline client/src/app/shared/video/video-miniature.component.html]: You don't need `span`

File: client/src/app/shared/video/video-miniature.component.html (hunk 2 above)
Comment (rigelk) [reply client/src/app/shared/video/video-miniature.component.html]: It's an action button, not an info button.

Comment (Chocobozzz) [reply client/src/app/shared/video/video-miniature.component.html]: The idea is to have the information loaded before the user clicks on the button

Comment (rigelk) [reply client/src/app/shared/video/video-miniature.component.html]: Ah, I see, sorry. I thought you meant to open the dropdown on mouseover.

Comment (rigelk) [reply client/src/app/shared/video/video-miniature.component.html]: @Chocobozzz working like a charm!

**Comment by Chocobozzz** [comment]
Closing in favour of https://github.com/Chocobozzz/PeerTube/commit/3a0fb65c61f80b510bce979a45d59d17948745e8

**Pull request closed without merging**